File: rb_ref_client/twister_client.rb
require 'socket'

class Twister
  def initialize(host, port)
    @host = host
    @port = port
  end

  def log
    @socket = UDPSocket.open
    @socket.connect(@host, @port)
    t=Time.now.to_i
    msg=[t].pack("Q")
    myclock=[1].pack("Q")
    myid="a4".to_a.pack("a50")
    mycomment="comment".to_a.pack("a50")
    mymsg=[]
    mymsg << msg.reverse
    mymsg << myclock.reverse
    mymsg << myid
    mymsg << mycomment
    puts mymsg.inspect
    puts mymsg.join("").length
    puts mymsg.join("")
    @socket.send(mymsg.join(""), 0, @host, @port)
  end
end

client = Twister.new("localhost", 4000)
client.log

File: python_ref_client/twister_client.py
from socket import *
import string
import time
import array
import struct
host = "127.0.0.1"
port = 4000
UDPSock = socket(AF_INET,SOCK_DGRAM)
timestamp= struct.pack('Q',int(time.time()))[::-1]
lc=struct.pack('Q',1)[::-1]
appid=struct.pack('!L',31)	
funid=struct.pack('!L',32)	
signalid=struct.pack('!L',33)	
myid=struct.pack('50s',"p5")
comment=struct.pack('50s',"python comment")

UDPSock.sendto(timestamp+lc+appid+funid+ signalid +myid+comment, (host,port))
